File: Samples/Evaluate/how-to-replace-standard-spreadsheetcontrol-command-with-your-own-custom-command-t163272-14.1.7-/VB/SpreadsheetCustomCommand/CustomService.vb
Imports DevExpress.XtraSpreadsheet
Imports DevExpress.XtraSpreadsheet.Commands
Imports DevExpress.XtraSpreadsheet.Services

Namespace SpreadsheetCustomCommand
    #Region "#customservice"
    Public Class CustomService
        Inherits SpreadsheetCommandFactoryServiceWrapper

        Public Sub New(ByVal service As ISpreadsheetCommandFactoryService)
            MyBase.New(service)
        End Sub
        Public Property Control() As SpreadsheetControl

        Public Overrides Function CreateCommand(ByVal id As SpreadsheetCommandId) As SpreadsheetCommand
            If id = SpreadsheetCommandId.FormatClearContents OrElse id = SpreadsheetCommandId.FormatClearContentsContextMenuItem Then
                Return New CustomFormatClearContentsCommand(Control)
            End If

            Return MyBase.CreateCommand(id)
        End Function

    End Class
    #End Region ' #customservice
End Namespace

File: Samples/Evaluate/how-to-replace-standard-spreadsheetcontrol-command-with-your-own-custom-command-t163272-14.1.7-/CS/SpreadsheetCustomCommand/CustomService.cs
using DevExpress.XtraSpreadsheet;
using DevExpress.XtraSpreadsheet.Commands;
using DevExpress.XtraSpreadsheet.Services;

namespace SpreadsheetCustomCommand
{
    #region #customservice
    public class CustomService : SpreadsheetCommandFactoryServiceWrapper {
        public CustomService(ISpreadsheetCommandFactoryService service)
            : base(service) {
        }
        public SpreadsheetControl Control {get;set;}

        public override SpreadsheetCommand CreateCommand(SpreadsheetCommandId id)
        {
            if (id == SpreadsheetCommandId.FormatClearContents || id == SpreadsheetCommandId.FormatClearContentsContextMenuItem)
                return new CustomFormatClearContentsCommand(Control);

            return base.CreateCommand(id);
        }

    }
    #endregion #customservice
}
